internal/confgen/parser.go

```diff
@@ -407,13 +407,22 @@ func (sp *sheetParser) parseMapField(field *Field, msg protoreflect.Message, rc
 				kvs := append(rc.CellDebugKV(keyColName), xerrors.KeyPBFieldType, "vertical map")
 				return false, xerrors.WithMessageKV(err, kvs...)
 			}
-			var newMapValue protoreflect.Value
-			if reflectMap.Has(newMapKey) {
-				// check key uniqueness
-				if err := prop.CheckKeyUnique(field.opts.Prop, cell.Data, true); err != nil {
+			// value must be empty if key not present
+			if !keyPresent && reflectMap.Has(newMapKey) {
+				tempCheckMapValue := reflectMap.NewValue()
+				valuePresent, err := sp.parseFieldOptions(tempCheckMapValue.Message(), rc, prefix+field.opts.Name)
+				if err != nil {
 					kvs := append(rc.CellDebugKV(keyColName), xerrors.KeyPBFieldType, "vertical map")
-					return false, xerrors.WrapKV(err, kvs...)
+					return false, xerrors.WithMessageKV(err, kvs...)
+				}
+				if valuePresent {
+					kvs := append(rc.CellDebugKV(keyColName), xerrors.KeyPBFieldType, "vertical map")
+					return false, xerrors.WithMessageKV(xerrors.E2017(keyColName), kvs...)
```

Comment:
Use `field.fd.FullName()` and `field.fd.Kind().String()` to fill E2017 error info.